base/abstractarray.jl

```diff
@@ -1829,6 +1829,21 @@ dimensions for every new input array and putting zero blocks elsewhere. For exam
 `cat(matrices...; dims=(1,2))` builds a block diagonal matrix, i.e. a block matrix with
 `matrices[1]`, `matrices[2]`, ... as diagonal blocks and matching zero blocks away from the
 diagonal.
+
+# Examples
+```jldoctest
+julia> cat(dims=[1, 2], fill(1, 1, 1), fill(2, 2, 2))
+3×3 Matrix{Int64}:
+1  0  0
+0  2  2
+0  2  2
+
+
+julia> cat(dims=[1, 2], [1 2], [3 4])
+2×4 Matrix{Int64}:
+1  2  0  0
+0  0  3  4
```

Comment:
Should all examples show the block behaviour?

In another (closed) PR my not-quite-as-compact suggestion was:
```suggestion
julia> cat(ones(2,2), fill(pi,2), zeros(2,3,1); dims=2)
2×6×1 Array{Float64, 3}:
[:, :, 1] =
 1.0  1.0  3.14159  0.0  0.0  0.0
 1.0  1.0  3.14159  0.0  0.0  0.0
 
julia> cat([true], trues(2,2), trues(2,4); dims=(1,2))
5×7 Matrix{Bool}:
 1  0  0  0  0  0  0
 0  1  1  0  0  0  0
 0  1  1  0  0  0  0
 0  0  0  1  1  1  1
 0  0  0  1  1  1  1